File: javascript/leetcode/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    seen = {}
    for (var i = 0; i < nums.length; i++) {
        if (seen[nums[i]]) { return true }
        seen[nums[i]] = 1
    }
    return false
};

File: ruby/leetcode/contains-duplicate.rb
# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
    seen = {}
    for i in 0...nums.size()
        if seen[nums[i]] != nil
            return true
        end
        seen[nums[i]] = 1
    end
    return false
end
